### ISSUE TITLE
CloudStream.iter?

### ISSUE THREAD

**Issue by dsyme** [mbraceproject/MBrace.Core#34]
Is there any reason why there isn't a ``CloudStream.iter`` that accepts a ``T --> Cloud<unit>``?

**Comment by palladin** [comment]
I think it is possible to add an iter-like combinator, maybe with a type sig like 
```
CloudStream.iter : CloudStream<'T> -> ('T -> Local<unit>) -> Cloud<unit>
```
Btw, I'm thinking very seriously to hide completely the Nessos.Streams types (one less namespace to open) from the surface API of MBrace and then rename CloudStream to CloudFlow. For various reasons I don't want to change the name of Nessos.Streams but for MBrace it can be just an implementation detail.

**Comment by dsyme** [comment]
Both of those sound great. Agreed that hiding "Nessos.Streams" makes sense.

**Comment by krontogiannis** [comment]
:+1:  for hiding the Streams types and renaming to CloudFlow.

If an `iter` is going to be added, with `'T -> Local<unit>` parameter; would it be useful to add functions like
`CloudStream.mapCloudf : ('T -> Local<'R>) -> CloudStream<'T>` etc?

Regardless of the fact that ParStream is used in the implementation, would operators like these be useful?

**Comment by palladin** [comment]
Yep combinators like CloudSteam.xxxCloud could be useful. 
As an example 
```
|> CloudStream (fun x -> local { do! Cloud.Log x in return x }) // enable logging
```

**Comment by dsyme** [comment]
Yes, I just found myself needed that. I think I prefer the name ``mapLocal`` though.

**Comment by dsyme** [comment]
fixed in #45
